File: Python/deck.py
class Card():
    def __init__(self, value, suit):
        self.value = value
        self.suit = suit
        if (value == 1):
            self.rank = 'Ace'
        elif (value == 11):
            self.rank = 'Jack'
        elif (value == 12):
            self.rank = 'Queen'
        elif (value == 13):
            self.rank = 'King'
        else:
            self.rank = value

    def __repr__(self):
        return f"{self.suit[0]} {self.value}"

    def __str__(self):
        return f"{self.rank} of {self.suit}"

# class to represent a deck of cards

class Deck():
    def __init__(self):
        self.contents = []
        suits = ['hearts', 'diamonds', 'clubs', 'spades']
        values = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
        for value in values:
            for suit in suits:
                self.contents.append(Card(value,suit))

new_deck = Deck()
print(new_deck.contents)

File: C#/cards/Deck.cs
using System;
using System.Collections.Generic;

namespace cards
{
    public class Deck 
    {
        public List<Card> cards;
        public Deck()
        {
            this.cards = new List<Card>();
            string[] suits = {"Hearts", "Clubs", "Diamonds", "Spades"};
            string[] names = { "Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King" };
            int[] values = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13 };
            for (int i = 0; i < suits.Length; i++) 
            {
                for (int j = 0; j < names.Length; j++) 
                {
                    cards.Add(new Card(suits[i], names[j], values[j]));
                }
            }
        }
        public Card Deal()
        {
            Card temp = cards[0];
            cards.RemoveAt(0);
            return temp;
        }
        public void Shuffle()
        {
            List<Card> shuffleDeck = new List<Card>();
            while (cards.Count != 0) 
            {
                Random rand = new Random();
                int index = rand.Next(0,cards.Count);
                shuffleDeck.Add(cards[index]);
                cards.RemoveAt(index);
            }
            cards = shuffleDeck;
        }
        public void Reset()
        {
            List<Card> resetList = new List<Card>();
            string[] suits = {"Hearts", "Clubs", "Diamonds", "Spades"};
            string[] names = { "Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King" };
            int[] values = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13 };
            for (int i = 0; i < suits.Length; i++) 
            {
                for (int j = 0; j < names.Length; j++) 
                {
                    resetList.Add(new Card(suits[i], names[j], values[j]));
                }
            }
            cards = resetList;
        }
    }
}